File: network/src/test/java/capstone/network/client/TestRecipeService.java
package capstone.network.client;

import org.junit.Before;
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import java.util.List;
import java.util.Objects;

import javax.inject.Inject;

import capstone.network.client.testdi.DaggerTestNetworkComponent;

public class TestRecipeService
{
    @Inject RecipeService service;

    TestRecipeService()
    {
        DaggerTestNetworkComponent.create().inject(this);
        Objects.nonNull(service);
    }

    @Test
    void testAllRecipes()
    {
        List<Recipe> allRecipes = service.getAllRecipes().blockingFirst();
        assertTrue(allRecipes.size() > 0);
    }

}

File: network/src/test/java/capstone/network/client/TestRecipeService.kt
package capstone.network.client

import okhttp3.mockwebserver.MockResponse
import okhttp3.mockwebserver.MockWebServer
import org.junit.Test
import org.junit.jupiter.api.Assertions.*

class TestRecipeService
{
    private val server = MockWebServer().apply {
        start()
    }

    init {
        server.enqueue(MockResponse().setBody(""))
    }


    @Test
    fun testBaseUrl()
    {
//        val service: RecipeService = NetworkModule(server.url("/")).provideRetrofit().create(RecipeService::class.java)
    }
}